client/src/annoMatrix/annoMatrix.ts

```diff
@@ -468,22 +467,22 @@ export default abstract class AnnoMatrix {
     query: Query
   ): WhereCacheColumnLabels | [undefined] {
     /*
-Return cache keys for columns associated with this query.  May return
-[unknown] if no keys are known (ie, nothing is or was cached).
-*/
+    Return cache keys for columns associated with this query.  May return
+    [unknown] if no keys are known (ie, nothing is or was cached).
+    */
     return _whereCacheGet(this._whereCache, this.schema, field, query);
   }
 
   /**
    ** Private interfaces below.
    **/
-  _resolveCachedQueries(field: Field, queries: Query[]): LabelArray {
+  _resolveCachedQueries(field: Field, queries: Query[]): LabelType[] {
     return queries
       .map((query: Query) =>
-        // @ts-expect-error --- TODO revisit:
-        // `filter`: This expression is not callable.
+        // @ts-expect-error ts-migrate --- suppressing TS defect (https://github.com/microsoft/TypeScript/issues/44373).
+        // Compiler is complaining that expression is not callable on array union types. Remove suppression once fixed.
         _whereCacheGet(this._whereCache, this.schema, field, query).filter(
-          (cacheKey?: LabelType) =>
+          (cacheKey: LabelType) =>
             cacheKey !== undefined && this._cache[field].hasCol(cacheKey)
```

Comment:
No.  Please look at the definition of _whereCacheGet type signature.  An explicit `undefined` is returned and must be filtered out.

Reply:
Ah sorry didn't think this through 🤔 

If `cacheKey` can be `undefined`, then `(cacheKey: LabelType) =>` should be `(cacheKey: LabelType | undefined) =>` ? Because `LabelType` is never `undefined`

And it seems like TS allows passing `undefined` to a function that accepts an optional argument:

```ts
function foo(bar?: number) {
  console.log(bar);
}

foo(0); // good
foo(); // good
foo(undefined); // good
foo(null); // bad
```
So `(cacheKey?: LabelType) =>` would work here too?

Reply:
You are right that it should be `LabelType | undefined`.  But that has different semantics than an optional arg.  Remember, passing a parameter with value === undefined is different than passing no parameter.

In this particular case, both  would work, but the semantics are different.